skrub/_datetime_encoder.py

```diff
@@ -18,78 +22,457 @@
+TIME_LEVELS = list(WORD_TO_ALIAS)
+
+
+def _is_pandas_format_mixed_available():
+    pandas_version = pd.__version__
+    min_pandas_version = "2.0.0"
+    return parse_version(min_pandas_version) < parse_version(pandas_version)
+
+
+MIXED_FORMAT = "mixed" if _is_pandas_format_mixed_available() else None
+
+
+def to_datetime(
+    X,
+    errors="coerce",
+    unit=None,
+    **kwargs,
+):
+    """
+    Convert argument to datetime. Return the input if not datetime-parsable.
+
+    Augment :func:`pandas.to_datetime` by supporting dataframes
+    and 2d arrays inputs. It converts compatible columns to datetime, and
+    pass incompatible columns unchanged.
+
+    int, float, str, datetime, list, tuple, 1d array, and Series are defered to
+    :func:`pandas.to_datetime` directly.
+
+    Parameters
+    ----------
+    X : int, float, str, datetime, list, tuple, nd array, Series, DataFrame/dict-like
+        The object to convert to a datetime.
+
+    errors : {'ignore', 'raise', 'coerce'}, default 'coerce'
+        - If ``'raise'``, then invalid parsing will raise an exception.
+        - If ``'coerce'``, then invalid parsing will be set as ``NaT``.
+        Note that ``'ignore'`` is not used for dataframes, 2d arrays,
+        and series, and is used otherwise as in ``pd.to_datetime``.
+
+    unit : str, default None
+        Unused. Here for compatibility with :func:`pandas.to_datetime`.
+
+    **kwargs : key, value mappings
+        Other keyword arguments are passed down to
+        :func:`pandas.to_datetime`.
+
+    Returns
+    -------
+    datetime
+        Return type depends on input.
+        - dataframes, series and 2d arrays return the same type
+        - otherwise return the same output as :func:`pandas.to_datetime`.
+
+    See Also
+    --------
+    :func:`pandas.to_datetime`
+        Convert argument to datetime.
+
+    Examples
+    --------
+    >>> X = pd.DataFrame(dict(a=[1, 2], b=["2021-01-01", "2021-02-02"]))
+    >>> X = to_datetime(X)
+    >>> X.dtypes.to_list()
+    [dtype('int64'), dtype('<M8[ns]')]
+    """
+    kwargs["errors"] = errors
+
+    # dataframe
+    if hasattr(X, "__dataframe__"):
+        return _to_datetime_dataframe(X, **kwargs)
+
+    # series, this attribute is available since Pandas 2.1.0
+    elif hasattr(X, "__column_consortium_standard__"):
+        return _to_datetime_series(X, **kwargs)
+
+    # 2d array
+    elif isinstance(X, Iterable) and np.asarray(X).ndim == 2:
+        X = _to_datetime_2d_array(np.asarray(X), **kwargs)
+        return np.vstack(X).T
+
+    # 1d array
+    elif isinstance(X, Iterable) and np.asarray(X).ndim == 1:
+        return _to_datetime_1d_array(np.asarray(X), **kwargs)
+
+    # scalar or unknown type
+    elif np.asarray(X).ndim == 0:
+        return _to_datetime_scalar(X, **kwargs)
+
+    else:
+        raise TypeError(
+            "X must be a string, datetime, list, tuple, 1-d array, Series, "
+            f"2-d array or dataframe. Got {X=!r}."
+        )
```

Comment:
@jeromedockes support for 1d array and scalar, LMKWYT

Reply:
it looks great!